app/src/main/java/com/github/sdpteam15/polyevents/EventActivity.kt

```diff
@@ -60,7 +79,53 @@ class EventActivity : AppCompatActivity() {
         findViewById<ImageView>(R.id.img_event_logo).apply {
             //TODO : change image
         }
+
+        if (event.isLimitedEvent()) {
+            subscribeButton.visibility = View.VISIBLE
+            if (currentDatabase.currentUser != null
+                && event.getParticipants().contains(currentDatabase.currentUser!!.uid)) {
+                subscribeButton.setText(resources.getString(R.string.event_unsubscribe))
+            }
+        } else {
+            subscribeButton.visibility = View.GONE
+        }
     }
 
+    fun onClickEventSubscribe(view: View) {
+        if (currentDatabase.currentUser == null) {
+            showToast(resources.getString(R.string.toast_subscribe_warning), this)
+        } else if (event.getParticipants().contains(currentDatabase.currentUser!!.uid)){
+            unsubscribeFromEvent()
+        } else {
+            subscribeToEvent()
+        }
+    }
+
+    fun unsubscribeFromEvent() {
+        event.removeParticipant(currentDatabase.currentUser!!.uid)
+
+        currentDatabase.eventDatabase!!.updateEvents(event)
+        showToast(resources.getString(R.string.event_successfully_unsubscribed), this)
+
+        subscribeButton.setText(resources.getString(R.string.event_subscribe))
+    }
+
+    fun subscribeToEvent() {
+        try {
+            event.addParticipant(currentDatabase.currentUser!!.uid)
+
+            currentDatabase.eventDatabase!!.updateEvents(event)
+            showToast(resources.getString(R.string.event_successfully_subscribed), this)
+            subscribeButton.setText(resources.getString(R.string.event_unsubscribe))
+        } catch (e: MaxAttendeesException) {
+            showToast(resources.getString(R.string.event_subscribe_at_max_capacity), this)
+        }
+    }
+
+    // Refactored here for tests
+    companion object {
+        var obsEvent = Observable<Event>()
+        lateinit var event: Event
+    }
```

Comment:
Can you put this on top of the file ? I had trouble to find where the variable are set :)

Reply:
Sure :)